sqlx-mysql/src/options/mod.rs

```diff
@@ -350,8 +350,13 @@ impl MySqlConnectOptions {
     /// substitution).
     ///
     /// <https://mariadb.com/kb/en/sql-mode/>
-    pub fn no_engine_subsitution(mut self, flag_val: bool) -> Self {
-        self.no_engine_subsitution = flag_val;
+    #[deprecated = "renamed to .no_engine_substitution()"]
+    pub fn no_engine_subsitution(self, flag_val: bool) -> Self {
+        self.no_engine_substitution(flag_val)
+    }
+    
+    pub fn no_engine_substitution(mut self, flag_val: bool) -> Self {
```

Comment:
Now also add the documentation from above.
Currently, this would not be documented ^^
```suggestion
    /// Flag that enables or disables the `NO_ENGINE_SUBSTITUTION` sql_mode setting after
    /// connection.
    ///
    /// If not set, if the available storage engine specified by a `CREATE TABLE` is not available,
    /// a warning is given and the default storage engine is used instead.
    ///
    /// By default, this is `true` (`NO_ENGINE_SUBSTITUTION` is passed, forbidding engine
    /// substitution).
    ///
    /// <https://mariadb.com/kb/en/sql-mode/>
    pub fn no_engine_substitution(mut self, flag_val: bool) -> Self {
```

Reply:
Yeah, you can just move the documentation over. No reason to have it in two places.